### PR TITLE
Made a simpler example for use in lectures so students can enter thei…

### PR DESCRIPTION
**Pull request by jenschr**
…r student name and use this as part of topic. This also prevents everyone from throwing each other out due to using the same client name

### CODE REVIEW

**Comment by jenschr** [comment]
I'm using in lectures and if students don't change the Client name, they'll throw each other out of the server. This example let's the student change the userID so this is used for both Client name and as part of the topic (preventing overwrites of the same topic). Also removes anything no required that can distract from explaining PubSub. This works much better when 30 ppl tries the same example at once :)

**Comment by EdJoPaTo** [comment]
Not sure if more examples are better. Also the naming is not ideal. The currently existing examples are `DelayedMQTTClientConfiguration`, `SimpleMQTTClient` and `twoMQTTClientHandling`. The first and the last have explaining names. The second is just "simple" which basically explains nothing. Adding `BasicMqttClient` is also not that speaking. Personally I would either merge both examples into one or name them better.

Maybe your example could be names `MinimalMqttClient` which indicates that it is the minimal required code to get it running. Maybe the "simple" one should be renamed too :thinking: 

(Just my thoughts without looking much into it and I am also not the repo owner, so I am just throwing in random thoughts)

**Comment by jenschr** [comment]
@EdJoPaTo I completely agree. I have updated the name to MinimalMqttClient in the PR.

File: examples/MinimalMqttClient/MinimalMqttClient.ino [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  This shows the basic handling of MQTT and Wifi connection.
+  Once it connects successfully to a Wifi network and a MQTT broker,
+  it subscribes to a topic and send a message to it.
+*/
+
+#include "EspMQTTClient.h"
+
+// Change these to customize where data will go
+String topic = "topicname";
+char userID[] = "yourname";
```
Comment (EdJoPaTo) [inline examples/MinimalMqttClient/MinimalMqttClient.ino:12]: Here is `String` and `char[]` used.

Personally I like to use the `#define` for Texts as you can concat multiple ones at compile time then (`"foo" "bar"` will be `"foobar"`). But I also understand why the `#define` is not used when explaining C++.

It should be the same to be easier to get for the user as its both just some text.

File: examples/MinimalMqttClient/MinimalMqttClient.ino [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+  BasicMQTTClient.ino
```
Comment (EdJoPaTo) [inline examples/MinimalMqttClient/MinimalMqttClient.ino:2]:
```suggestion
  MinimalMQTTClient.ino
```

**Comment by plapointe6** [comment]
I understand your point, but the new MinimalExemple is really similar to the BasicMQTTClient exemple. I am agree with @EdJoPaTo more exemple does not mean better.

It's fun to know that this lib is used for education purposes :)
Should be really fun for students to play with ESP. 

PS: @EdJoPaTo thanks for your help.  I have less time than before to give on this lib and your help is really appreciated.

**Pull request closed without merging**